src/client/app/containers/view.mode.ts

```diff
@@ -0,0 +1,85 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { Subscription } from 'rxjs/Subscription';
+import * as fromRoot from '../reducers';
+import { UI, Snippet } from '../actions';
+import { environment, applyTheme } from '../helpers';
+import { Strings } from '../strings';
+
+@Component({
+    selector: 'view-mode',
+    template: `
+        <main [ngClass]="theme$|async">
+            <header class="command__bar">
+                <command class="view-mode" [hidden]="isEmpty" [title]="snippet?.name"></command>
+            </header>
+            <editor [isViewMode]="true"></editor>
+            <footer class="command__bar command__bar--condensed">
+                <command id="feedback" [title]="Feedback" icon="Emoji2" (click)="feedback()"></command>
+                <command icon="Color" [title]="theme$|async" (click)="changeTheme()"></command>
+            </footer>
+        </main>
+    `
+})
+
+
+export class ViewMode implements OnInit, OnDestroy {
+    strings = Strings();
+    snippet: ISnippet;
+    showInfo: boolean;
+    paramsSub: Subscription;
+
+    constructor(
+        private _store: Store<fromRoot.State>,
+        private _route: ActivatedRoute
+    ) {
+        this._store.select(fromRoot.getCurrent).subscribe(snippet => {
```

Comment:
Lets clean this up later because we do it in other places too, but we should always unsubscribe OnDestroy or this subscription will hold a reference to ViewMode and it will leak ViewMode object.